packages/storage/src/userStorage.ts

```diff
@@ -326,10 +336,15 @@ export class UserStorage {
     return summary;
   }
 
-  private async getOrCreateBucket(client: Buckets, name: string): Promise<GetOrCreateResponse> {
-    return client.getOrCreate(name, {
-      threadID: getDeterministicThreadID(this.user.identity as PrivateKey).toString(),
-    });
+  private async getOrCreateBucket(client: Buckets, name: string): Promise<BucketMetadataWithThreads> {
+    const threadId = getDeterministicThreadID(this.user.identity as PrivateKey).toString();
+    const metadataStore = await this.getMetadataStore();
+    const metadata: BucketMetadata = await metadataStore.findBucket(name, threadId)
+      || await metadataStore.createBucket(name, threadId);
+
+    const getOrCreateResponse = await client.getOrCreate(name, { threadID: threadId });
```

Comment:
Is the idea here that we would use the same thread Id for all the buckets? In the daemon we did a new threadId for each bucket and this also means if someone shares the thread key they share it for all buckets.  It should be fine either way but just highlighting it in case we should discuss.

Reply:
Hmmm... I'll have to look at the daemon once more. I thought threadId's were deterministic (based on the users identity) and that was what was used with buckets.

Reply:
Okay, I see what you mean. The deterministic id was to store the users metadata in threadsdb, but each bucket entry had its own threadId generated. I will update the `BucketMetadata` fields to have a threadId per bucket and use that results here.